# Investment Assignment

Data Analysis to find the best suitable Investment Horizon for **Spark Funds**

## 1. Investment Type Analysis

Finding the Best Investment Type for Spark Funds

In [1]:
import pandas

Loading the **rounds2.csv** file using pandas. 
The length of the column **company_permalink** must be the number of unique companies in it.

In [2]:
rounds2 = pandas.read_csv('rounds2.csv', sep=',', encoding='unicode_escape')
rounds2_permalinks = rounds2['company_permalink'].str.lower()
rounds2_unique_permalinks = set(rounds2_permalinks)
print(f'Length of rounds2 Unique permalink : {len(rounds2_unique_permalinks)}')

Length of rounds2 Unique permalink : 66370


Repeating the same steps above for **companies.csv**

In [3]:
companies = pandas.read_csv('companies.csv', sep='\t', encoding='unicode_escape')
companies_permalinks = companies['permalink'].str.lower()
companies_unique_permalinks = set(companies_permalinks)
print(f'Length of comapnies Unique permalinks : {len(companies_unique_permalinks)}')

Length of comapnies Unique permalinks : 66368


In [4]:
difference = rounds2_unique_permalinks.difference(companies_unique_permalinks)   # set difference
print(f'Number of companies that are NOT in companies.csv but in rounds2.csv = {len(difference)}')

Number of companies that are NOT in companies.csv but in rounds2.csv = 57


Converting every value to lower case of column **company_permalink** of rounds2 and renaming as **permalink**

In [5]:
rounds2['company_permalink'] = rounds2['company_permalink'].str.lower()
rounds2 = rounds2.rename({'company_permalink': 'permalink'}, axis=1)
rounds2.head()

permalink  \
0                  /organization/-fame   
1               /organization/-qounter   
2               /organization/-qounter   
3  /organization/-the-one-of-them-inc-   
4                /organization/0-6-com   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  A  14-10-2014                NaN  
2                NaN  01-03-2014           700000.0  
3                  B  30-01-2014          3406878.0  
4                  A  19-03-2008          2000000.0

Converting every value to lower case of column **permalink** of companies.

In [6]:
companies['permalink'] = companies['permalink'].str.lower()
companies.head()

permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
2  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
3                /organization/0-6-com                 0-6.com   
4       /organization/004-technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010

Merging the two dataframes together as **master**

In [7]:
companies.set_index(['permalink'])
rounds2.set_index(['permalink'])
master = pandas.merge(companies, rounds2,how='inner', on='permalink')
master.head()

permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
2               /organization/-qounter                :Qounter   
3  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
4                /organization/0-6-com                 0-6.com   

             homepage_url                                      category_list  \
0      http://livfame.com                                              Media   
1  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
2  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
3     http://oneofthem.jp                                  Apps|Games|Mobile   
4      http://www.0-6.com                                        Curated Web   

      status country_code state_code      region           city  founded_at  \
0  operating          IND         16      Mumbai         Mumbai         NaN   
1  operating          USA         DE  DE - Other  Delaware City  04-09-2014   
2  operating          USA         DE  DE - Other  Delaware City  04-09-2014   
3  operating          NaN        NaN         NaN            NaN         NaN   
4  operating          CHN         22     Beijing        Beijing  01-01-2007   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  A  14-10-2014                NaN  
2                NaN  01-03-2014           700000.0  
3                  B  30-01-2014          3406878.0  
4                  A  19-03-2008          2000000.0

In [8]:
print(f'Number of records in master dataframe = {len(master)}')

Number of records in master dataframe = 114888


Finding **The most representing value** for the following fund types.
1. Venture fund
2. Angel fund
3. Seed fund
4. Private Equity fund.

Note : All values are rounded off to two decimal places.

In [9]:
venture_fund_mean = master.loc[master.funding_round_type=='venture'].raised_amount_usd.mean()
angel_fund_mean = master.loc[master.funding_round_type=='angel'].raised_amount_usd.mean()
seed_fund_mean = master.loc[master.funding_round_type=='seed'].raised_amount_usd.mean()
private_equity_fund_mean = master.loc[master.funding_round_type=='private_equity'].raised_amount_usd.mean()

print(f'Venture fund average : {round(venture_fund_mean, 2)}')
print(f'Angel fund average : {round(angel_fund_mean, 2)}')
print(f'Seed fund average : {round(seed_fund_mean, 2)}')
print(f'Private Equity fund average : {round(private_equity_fund_mean, 2)}')

Venture fund average : 11749430.94
Angel fund average : 958694.47
Seed fund average : 719719.06
Private Equity fund average : 73341461.74


**Since Spark Funds is ready to Invest between 5 to 10 million USD, _Venture Funding Type_ looks most suitable for them**

## 2. Country Analysis

From now, We'll use only venture type investment data for further analysis.

In [10]:
master = master.loc[master.funding_round_type=='venture']
master.head()

permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
3  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
4                /organization/0-6-com                 0-6.com   
5       /organization/004-technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
3                http://oneofthem.jp   
4                 http://www.0-6.com   
5  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
3                                  Apps|Games|Mobile  operating          NaN   
4                                        Curated Web  operating          CHN   
5                                           Software  operating          USA   

  state_code                 region           city  founded_at  \
0         16                 Mumbai         Mumbai         NaN   
1         DE             DE - Other  Delaware City  04-09-2014   
3        NaN                    NaN            NaN         NaN   
4         22                Beijing        Beijing  01-01-2007   
5         IL  Springfield, Illinois      Champaign  01-01-2010   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5  /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  A  14-10-2014                NaN  
3                  B  30-01-2014          3406878.0  
4                  A  19-03-2008          2000000.0  
5                NaN  24-07-2014                NaN

We'll now do a quality check on the data. Let's find out the percentage of rows with **raised_amount_usd** value missing.

In [11]:
percent = round((master.raised_amount_usd.isnull().sum()/master.raised_amount_usd.count()) * 100, 3)
print(f'Percentage of missing values = {percent}%')

Percentage of missing values = 10.474%


Since only **10.4%** of the rows has missing values, the data is still good to use. But to deal with the missing values, we'll fill them up with mean value of **raied_amount_usd**

**The reason why we didn't dealt with the missing values before is that, before this we were dealing with data of all kinds of Investments link venture type, seed type etc. Therefore filling the missing values with a mean/median value didn't seem to agree with a real world scenario.**

In [12]:
mean_raised_amount = master.raised_amount_usd.mean()
master['raised_amount_usd'] = master.raised_amount_usd.fillna(mean_raised_amount)
master.head()

permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
3  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
4                /organization/0-6-com                 0-6.com   
5       /organization/004-technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
3                http://oneofthem.jp   
4                 http://www.0-6.com   
5  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
3                                  Apps|Games|Mobile  operating          NaN   
4                                        Curated Web  operating          CHN   
5                                           Software  operating          USA   

  state_code                 region           city  founded_at  \
0         16                 Mumbai         Mumbai         NaN   
1         DE             DE - Other  Delaware City  04-09-2014   
3        NaN                    NaN            NaN         NaN   
4         22                Beijing        Beijing  01-01-2007   
5         IL  Springfield, Illinois      Champaign  01-01-2010   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5  /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015       1.000000e+07  
1                  A  14-10-2014       1.174943e+07  
3                  B  30-01-2014       3.406878e+06  
4                  A  19-03-2008       2.000000e+06  
5                NaN  24-07-2014       1.174943e+07

The Top 9 Countries with **Highest Total Venture Funding** is listed below.

In [13]:
highest_funded_countries = master.groupby(['country_code'])['raised_amount_usd'].sum()
highest_funded_countries = highest_funded_countries.sort_values(ascending=False).iloc[:9]
highest_funded_countries

country_code
USA    4.487473e+11
CHN    4.598037e+10
GBR    2.315949e+10
IND    1.636576e+10
CAN    1.064078e+10
DEU    9.024430e+09
ISR    8.246950e+09
FRA    8.040104e+09
JPN    3.880652e+09
Name: raised_amount_usd, dtype: float64

Creating a new dataframe **top9** with details of Venture funding data in the top 9 countries.

In [14]:
top9 = master.loc[master['country_code'].apply(lambda x: x in highest_funded_countries)]
top9.head()

permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
4                /organization/0-6-com                 0-6.com   
5       /organization/004-technologies        004 Technologies   
8  /organization/0ndine-biomedical-inc  Ondine Biomedical Inc.   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
4                 http://www.0-6.com   
5  http://004gmbh.de/en/004-interact   
8               http://ondinebio.com   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
4                                        Curated Web  operating          CHN   
5                                           Software  operating          USA   
8                                      Biotechnology  operating          CAN   

  state_code                 region           city  founded_at  \
0         16                 Mumbai         Mumbai         NaN   
1         DE             DE - Other  Delaware City  04-09-2014   
4         22                Beijing        Beijing  01-01-2007   
5         IL  Springfield, Illinois      Champaign  01-01-2010   
8         BC              Vancouver      Vancouver  01-01-1997   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5  /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   
8  /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015       1.000000e+07  
1                  A  14-10-2014       1.174943e+07  
4                  A  19-03-2008       2.000000e+06  
5                NaN  24-07-2014       1.174943e+07  
8                NaN  21-12-2009       7.194910e+05

 The List of English Speaking countries were collected from **[this webpage](https://en.wikipedia.org/wiki/List_of_territorial_entities_where_English_is_an_official_language)**.

And we the top 3 english speaking countries are
1. **USA**
2. **GBR**
3. **CAN**

## 3. Sector Analysis

Let's read the **mapping.csv** file into a dataframe and see how it looks.

In [15]:
mapping_raw = pandas.read_csv('mapping.csv', index_col='category_list')
mapping_raw.head()

Automotive & Sports  Blanks  Cleantech / Semiconductors  \
category_list                                                            
NaN                              0       1                           0   
3D                               0       0                           0   
3D Printing                      0       0                           0   
3D Technology                    0       0                           0   
Accounting                       0       0                           0   

               Entertainment  Health  Manufacturing  \
category_list                                         
NaN                        0       0              0   
3D                         0       0              1   
3D Printing                0       0              1   
3D Technology              0       0              1   
Accounting                 0       0              0   

               News, Search and Messaging  Others  \
category_list                                       
NaN                                     0       0   
3D                                      0       0   
3D Printing                             0       0   
3D Technology                           0       0   
Accounting                              0       0   

               Social, Finance, Analytics, Advertising  
category_list                                           
NaN                                                  0  
3D                                                   0  
3D Printing                                          0  
3D Technology                                        0  
Accounting                                           1

Let's reverse the one-hot encoding for an easier analysis

In [16]:
mapping = pandas.DataFrame(pandas.get_dummies(mapping_raw).idxmax(1))
mapping.columns = ['main_sector']
mapping.index.names = ['primary_sector']
mapping.head()

main_sector
primary_sector                                         
NaN                                              Blanks
3D                                        Manufacturing
3D Printing                               Manufacturing
3D Technology                             Manufacturing
Accounting      Social, Finance, Analytics, Advertising

Since we've discussed with the CEO and formed a business rule that **the first string before the vertical bar will be considered the primary sector**, we'll ignore the rest of the values in the **category_list** column in the master dataframe. We'll edit the master dataframe such that the primary sector according to the business rule will be there in the new column **primary_sector**.

In [17]:
master.category_list = master.category_list.astype(str)
master['primary_sector'] = master.category_list.apply(lambda x: x.split('|')[0])
master.head()

permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
3  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
4                /organization/0-6-com                 0-6.com   
5       /organization/004-technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
3                http://oneofthem.jp   
4                 http://www.0-6.com   
5  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
3                                  Apps|Games|Mobile  operating          NaN   
4                                        Curated Web  operating          CHN   
5                                           Software  operating          USA   

  state_code                 region           city  founded_at  \
0         16                 Mumbai         Mumbai         NaN   
1         DE             DE - Other  Delaware City  04-09-2014   
3        NaN                    NaN            NaN         NaN   
4         22                Beijing        Beijing  01-01-2007   
5         IL  Springfield, Illinois      Champaign  01-01-2010   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5  /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   

  funding_round_code   funded_at  raised_amount_usd         primary_sector  
0                  B  05-01-2015       1.000000e+07                  Media  
1                  A  14-10-2014       1.174943e+07  Application Platforms  
3                  B  30-01-2014       3.406878e+06                   Apps  
4                  A  19-03-2008       2.000000e+06            Curated Web  
5                NaN  24-07-2014       1.174943e+07               Software

Now let's merge the master data with the mapping data

In [18]:
master = pandas.merge(master, mapping, how='inner', on='primary_sector')
master.head(10)

permalink             name              homepage_url  \
0            /organization/-fame            #fame        http://livfame.com   
1            /organization/90min            90min      http://www.90min.com   
2            /organization/90min            90min      http://www.90min.com   
3            /organization/90min            90min      http://www.90min.com   
4  /organization/all-def-digital  All Def Digital  http://alldefdigital.com   
5      /organization/audiosocket      Audiosocket    http://audiosocket.com   
6      /organization/audiosocket      Audiosocket    http://audiosocket.com   
7      /organization/audiosocket      Audiosocket    http://audiosocket.com   
8           /organization/babble           Babble     http://www.babble.com   
9           /organization/babble           Babble     http://www.babble.com   

                         category_list     status country_code state_code  \
0                                Media  operating          IND         16   
1  Media|News|Publishing|Soccer|Sports  operating          GBR         H9   
2  Media|News|Publishing|Soccer|Sports  operating          GBR         H9   
3  Media|News|Publishing|Soccer|Sports  operating          GBR         H9   
4                                Media  operating          USA         CA   
5               Media|Music|Technology  operating          USA         WA   
6               Media|Music|Technology  operating          USA         WA   
7               Media|Music|Technology  operating          USA         WA   
8                 Media|News|Parenting   acquired          USA         NY   
9                 Media|News|Parenting   acquired          USA         NY   

          region         city  founded_at  \
0         Mumbai       Mumbai         NaN   
1         London       London  01-01-2011   
2         London       London  01-01-2011   
3         London       London  01-01-2011   
4    Los Angeles  Los Angeles         NaN   
5        Seattle      Seattle  01-05-2008   
6        Seattle      Seattle  01-05-2008   
7        Seattle      Seattle  01-05-2008   
8  New York City     New York  01-12-2006   
9  New York City     New York  01-12-2006   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6            venture   
2  /funding-round/bd626ed022f5c66574b1afe234f3c90d            venture   
3  /funding-round/fd4b15e8c97ee2ffc0acccdbe1a98810            venture   
4  /funding-round/452a2342fe720285c3b92e9bd927d9ba            venture   
5  /funding-round/0959f59e3a89bc8db8c6ed9c646349fd            venture   
6  /funding-round/4e3a53eec6c609943df752206070b23f            venture   
7  /funding-round/bc0b412cf2fa7b739e4105ad756b47a4            venture   
8  /funding-round/43cc678b39be3e684bb88501af05ff9b            venture   
9  /funding-round/5fdb3dbeeb6eea0243702adc372130f3            venture   

  funding_round_code   funded_at  raised_amount_usd primary_sector  \
0                  B  05-01-2015         10000000.0          Media   
1                NaN  06-10-2015         15000000.0          Media   
2                NaN  07-05-2013          5800000.0          Media   
3                NaN  26-03-2014         18000000.0          Media   
4                  A  06-08-2014          5000000.0          Media   
5                NaN  02-05-2013           650000.0          Media   
6                  A  02-09-2009           750000.0          Media   
7                NaN  15-02-2012          1582300.0          Media   
8                  A  05-02-2009          2000000.0          Media   
9                  A  12-05-2009          1250000.0          Media   

     main_sector  
0  Entertainment  
1  Entertainment  
2  Entertainment  
3  Entertainment  
4  Entertainment  
5  Entertainment  
6  Entertainment  
7  Entertainment  
8  Entertainment  
9  Entertainment

Now we have the master dataframe with two additional columns **primary_sector** and **main_sector**. We can do detailed analysis with respect to sectors which the startups work in.

Creating 3 dataframes with data of only the top 3 English speaking countries. ie;
1. USA
2. GBR
3. CAN

Filtering data for **USA**

In [19]:
usa_funding = master.loc[master.country_code=='USA']
usa_funding.head()

permalink             name              homepage_url  \
4  /organization/all-def-digital  All Def Digital  http://alldefdigital.com   
5      /organization/audiosocket      Audiosocket    http://audiosocket.com   
6      /organization/audiosocket      Audiosocket    http://audiosocket.com   
7      /organization/audiosocket      Audiosocket    http://audiosocket.com   
8           /organization/babble           Babble     http://www.babble.com   

            category_list     status country_code state_code         region  \
4                   Media  operating          USA         CA    Los Angeles   
5  Media|Music|Technology  operating          USA         WA        Seattle   
6  Media|Music|Technology  operating          USA         WA        Seattle   
7  Media|Music|Technology  operating          USA         WA        Seattle   
8    Media|News|Parenting   acquired          USA         NY  New York City   

          city  founded_at                          funding_round_permalink  \
4  Los Angeles         NaN  /funding-round/452a2342fe720285c3b92e9bd927d9ba   
5      Seattle  01-05-2008  /funding-round/0959f59e3a89bc8db8c6ed9c646349fd   
6      Seattle  01-05-2008  /funding-round/4e3a53eec6c609943df752206070b23f   
7      Seattle  01-05-2008  /funding-round/bc0b412cf2fa7b739e4105ad756b47a4   
8     New York  01-12-2006  /funding-round/43cc678b39be3e684bb88501af05ff9b   

  funding_round_type funding_round_code   funded_at  raised_amount_usd  \
4            venture                  A  06-08-2014          5000000.0   
5            venture                NaN  02-05-2013           650000.0   
6            venture                  A  02-09-2009           750000.0   
7            venture                NaN  15-02-2012          1582300.0   
8            venture                  A  05-02-2009          2000000.0   

  primary_sector    main_sector  
4          Media  Entertainment  
5          Media  Entertainment  
6          Media  Entertainment  
7          Media  Entertainment  
8          Media  Entertainment

Finding the **total number of investments** in **USA**

In [20]:
len(usa_funding.main_sector)

35421

The **total number of investments** in **USA** = **35421**

Finding the **total amount invested** in **USA** 

In [21]:
round(usa_funding.raised_amount_usd.sum(), 2)

414502753027.06

The **total amount invested** in **USA** = $ **414502753027.06**

Finding the **number of investments** in each sector in **USA**

In [22]:
usa_funding.groupby(['main_sector']).size().sort_values(ascending=False)

main_sector
Others                                     8693
Cleantech / Semiconductors                 8113
Social, Finance, Analytics, Advertising    5553
News, Search and Messaging                 4562
Health                                     3403
Manufacturing                              2643
Entertainment                              1888
Automotive & Sports                         566
dtype: int64

The top 3 sectors with most number of investments in **USA** are : 
1. **Others**
2. **Cleantech / Semiconductors**
3. **Social, Finance, Analytics, Advertising**

Finding the **Company that received highest investment** in **Others** Sector in **USA**.

In [23]:
usa_funding.loc[usa_funding.main_sector == 'Others'].sort_values('raised_amount_usd', ascending=False)[['permalink', 'name', 'country_code', 'funding_round_type', 'main_sector', 'raised_amount_usd']][:1]

permalink  name country_code funding_round_type  \
14280  /organization/social-finance  SoFi          USA            venture   

      main_sector  raised_amount_usd  
14280      Others       1.000000e+09

Looks like the company **SoFi (/organization/social-finance)** received the highest investment of **$1000000000** in **Others** Sector in **USA**

Finding the **Company that received highest investment** in **Cleantech / Semiconductors** Sector in **USA**.

In [24]:
usa_funding.loc[usa_funding.main_sector == 'Cleantech / Semiconductors'].sort_values('raised_amount_usd', ascending=False)[['permalink', 'name', 'country_code', 'funding_round_type', 'main_sector', 'raised_amount_usd']][:1]

permalink                     name country_code  \
36981  /organization/freescale  Freescale Semiconductor          USA   

      funding_round_type                 main_sector  raised_amount_usd  
36981            venture  Cleantech / Semiconductors       1.760000e+10

Looks like the company **Freescale Semiconductor (/organization/freescale)** received the highest investment of **$17600000000** in **Cleantech / Semiconductors** Sector in **USA**

Repeating same steps above for **GBR**

In [25]:
gbr_funding = master.loc[master.country_code=='GBR']
gbr_funding.head()

permalink                         name  \
1                          /organization/90min                        90min   
2                          /organization/90min                        90min   
3                          /organization/90min                        90min   
66                        /organization/teepix                       Teepix   
102  /organization/common-interest-communities  Common Interest Communities   

                              homepage_url  \
1                     http://www.90min.com   
2                     http://www.90min.com   
3                     http://www.90min.com   
66                                     NaN   
102  http://commoninterestcommunities.com/   

                                        category_list     status country_code  \
1                 Media|News|Publishing|Soccer|Sports  operating          GBR   
2                 Media|News|Publishing|Soccer|Sports  operating          GBR   
3                 Media|News|Publishing|Soccer|Sports  operating          GBR   
66                Media|Photography|Service Providers  operating          GBR   
102  Application Platforms|Internet|Software|Startups  operating          GBR   

    state_code  region     city  founded_at  \
1           H9  London   London  01-01-2011   
2           H9  London   London  01-01-2011   
3           H9  London   London  01-01-2011   
66          A5  London  Bedford         NaN   
102         H9  London   London         NaN   

                             funding_round_permalink funding_round_type  \
1    /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6            venture   
2    /funding-round/bd626ed022f5c66574b1afe234f3c90d            venture   
3    /funding-round/fd4b15e8c97ee2ffc0acccdbe1a98810            venture   
66   /funding-round/71b43b36b581697e603b256a13229994            venture   
102  /funding-round/8195587cbd5e51af7514ee92ef4ba6ba            venture   

    funding_round_code   funded_at  raised_amount_usd         primary_sector  \
1                  NaN  06-10-2015         15000000.0                  Media   
2                  NaN  07-05-2013          5800000.0                  Media   
3                  NaN  26-03-2014         18000000.0                  Media   
66                 NaN  14-11-2007          1655405.0                  Media   
102                NaN  09-07-2014         10000000.0  Application Platforms   

                    main_sector  
1                 Entertainment  
2                 Entertainment  
3                 Entertainment  
66                Entertainment  
102  News, Search and Messaging

Finding the **total number of investments** in **GBR**

In [26]:
len(gbr_funding.main_sector)

2120

The **total number of investments** in **GBR** = **2120**

Finding the **total amount invested** in **GBR** 

In [27]:
round(gbr_funding.raised_amount_usd.sum(), 2)

21321274230.84

The **total amount invested** in **GBR** = $**21321274230.84**

Finding the **number of investments** in each sector in **GBR**

In [28]:
gbr_funding.groupby(['main_sector']).size().sort_values(ascending=False)

main_sector
Others                                     571
Cleantech / Semiconductors                 456
Social, Finance, Analytics, Advertising    359
News, Search and Messaging                 265
Entertainment                              157
Manufacturing                              132
Health                                     126
Automotive & Sports                         54
dtype: int64

The top 3 sectors with most number of investments in **GBR** are : 
1. **Others**
2. **Cleantech / Semiconductors**
3. **Social, Finance, Analytics, Advertising**

Which is same as **USA**

Finding the **Company that received highest investment** in **Others** Sector in **GBR**.

In [29]:
gbr_funding.loc[gbr_funding.main_sector == 'Others'].sort_values('raised_amount_usd', ascending=False)[['permalink', 'name', 'country_code', 'funding_round_type', 'main_sector', 'raised_amount_usd']][:1]

permalink    name country_code funding_round_type  \
36036  /organization/oneweb  OneWeb          GBR            venture   

      main_sector  raised_amount_usd  
36036      Others        500000000.0

Looks like the company **OneWeb (/organization/oneweb)** received the highest investment of **$ 500000000** in **Others** Sector in **GBR**

Finding the **Company that received highest investment** in **Cleantech / Semiconductors** Sector in **GBR**.

In [30]:
gbr_funding.loc[gbr_funding.main_sector == 'Cleantech / Semiconductors'].sort_values('raised_amount_usd', ascending=False)[['permalink', 'name', 'country_code', 'funding_round_type', 'main_sector', 'raised_amount_usd']][:1]

permalink        name country_code funding_round_type  \
10189  /organization/immunocore  Immunocore          GBR            venture   

                      main_sector  raised_amount_usd  
10189  Cleantech / Semiconductors        320000000.0

Looks like the company **Immunocore (/organization/immunocore)** received the highest investment of **$320000000** in **Cleantech / Semiconductors** Sector in **GBR**

Repeating same steps above for **CAN**

In [31]:
can_funding = master.loc[master.country_code=='CAN']
can_funding.head()

permalink          name            homepage_url  \
136  /organization/mmb-networks  MMB Networks  http://mmbnetworks.com   
184         /organization/500px         500px        http://500px.com   
185         /organization/500px         500px        http://500px.com   
186         /organization/500px         500px        http://500px.com   
187         /organization/500px         500px        http://500px.com   

                                         category_list     status  \
136  Application Platforms|Customer Service|Embedde...  operating   
184  Apps|Internet|Photography|Promotional|Sales an...  operating   
185  Apps|Internet|Photography|Promotional|Sales an...  operating   
186  Apps|Internet|Photography|Promotional|Sales an...  operating   
187  Apps|Internet|Photography|Promotional|Sales an...  operating   

    country_code state_code   region     city  founded_at  \
136          CAN         ON  Toronto  Toronto  01-01-2008   
184          CAN         ON  Toronto  Toronto  01-10-2009   
185          CAN         ON  Toronto  Toronto  01-10-2009   
186          CAN         ON  Toronto  Toronto  01-10-2009   
187          CAN         ON  Toronto  Toronto  01-10-2009   

                             funding_round_permalink funding_round_type  \
136  /funding-round/7c00988bec2b2d5a0c2ed95636b0ee05            venture   
184  /funding-round/344b1e2b60ff76f4548c69ea7c04901f            venture   
185  /funding-round/36b2c6975d20b602b5d9ea9724e383ea            venture   
186  /funding-round/79c5fa844058b277cfa4d14231e71c9e            venture   
187  /funding-round/8d77dbdb6825f2f026713e49bc855b34            venture   

    funding_round_code   funded_at  raised_amount_usd         primary_sector  \
136                  A  01-12-2013       3.000000e+06  Application Platforms   
184                  A  07-08-2013       8.800000e+06                   Apps   
185                  B  21-07-2015       1.300000e+07                   Apps   
186                  A  01-10-2012       1.174943e+07                   Apps   
187                  A  03-06-2011       5.250000e+05                   Apps   

                    main_sector  
136  News, Search and Messaging  
184  News, Search and Messaging  
185  News, Search and Messaging  
186  News, Search and Messaging  
187  News, Search and Messaging

Finding the **total number of investments** in **CAN**

In [32]:
len(can_funding.main_sector)

1256

The **total number of investments** in **CAN** = **1256**

Finding the **total amount invested** in **CAN** 

In [33]:
round(can_funding.raised_amount_usd.sum(), 2)

9655574999.23

The **total amount invested** in **CAN** = $**9655574999.23**

Finding the **number of investments** in each sector in **CAN**

In [34]:
can_funding.groupby(['main_sector']).size().sort_values(ascending=False)

main_sector
Cleantech / Semiconductors                 346
Others                                     337
Social, Finance, Analytics, Advertising    175
News, Search and Messaging                 145
Manufacturing                              113
Health                                      65
Entertainment                               58
Automotive & Sports                         17
dtype: int64

The top 3 sectors with most number of investments in USA are : 
1. **Cleantech / Semiconductors**
2. **Others**
3. **Social, Finance, Analytics, Advertising**

The top 3 sectors are still the same as if **USA** and **CAN** but **Cleantech / Semiconductors** and **Others** sectors interchanged positions.

Finding the **Company that received highest investment** in **Cleantech / Semiconductors** Sector in **CAN**.

In [35]:
can_funding.loc[can_funding.main_sector == 'Cleantech / Semiconductors'].sort_values('raised_amount_usd', ascending=False)[['permalink', 'name', 'country_code', 'funding_round_type', 'main_sector', 'raised_amount_usd']][:1]

permalink       name country_code funding_round_type  \
28850  /organization/ram-power  Ram Power          CAN            venture   

                      main_sector  raised_amount_usd  
28850  Cleantech / Semiconductors         77000000.0

Looks like the company **Ram Power (/organization/ram-power)** received the highest investment of **$77000000** in **Cleantech / Semiconductors** Sector in **CAN**

Finding the **Company that received highest investment** in **Others** Sector in **CAN**.

In [36]:
can_funding.loc[can_funding.main_sector == 'Others'].sort_values('raised_amount_usd', ascending=False)[['permalink', 'name', 'country_code', 'funding_round_type', 'main_sector', 'raised_amount_usd']][:1]

permalink name country_code funding_round_type  \
31361  /organization/desire2learn  D2L          CAN            venture   

      main_sector  raised_amount_usd  
31361      Others         85000000.0

Looks like the company **D2L (/organization/desire2learn)** received the highest investment of **$85000000** in **Others** Sector in **CAN**